src/basic/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Basic Auth plugin.
+ */
+class BasicAuth {
+  constructor({ loginPath = '/login', logoutPath = '/logout', fetchUser = null } = {}) {
+    this.loginPath = loginPath;
+    this.logoutPath = logoutPath;
+    this.fetchUser = fetchUser;
+
+    if (!this.fetchUser) {
+      throw new Error('fetchUser is required');
+    }
+  }
+
+  init() {
+    return this.fetchUser()
```

Comment:
Should we require `fetchUser` is a function?

Reply:
yes!